Src/IronPython/Runtime/ArrayData.cs

```diff
@@ -102,18 +102,30 @@ internal class ArrayData<T> : ArrayData, IList<T>, IReadOnlyList<T> where T : st
         }
 
         public void AddRange(IPythonBuffer data) {
-            ReadOnlySpan<byte> dataSpan = data.AsReadOnlySpan();
+            if (data.IsCContiguous()) {
```

Comment:
Same here, seems like you could hide the details inside a `CopyTo(IPythonBuffer, Span<byte>)` extension?

Reply:
OK.